src/core/PathTracingSceneGenerator.js

```diff
@@ -176,9 +178,29 @@ export class PathTracingSceneGenerator {
 		// generate the geometry
 		const result = staticGeometryGenerator.generate( geometry );
 		const materials = result.materials;
+		let needsMaterialIndexUpdate = result.changeType !== NO_CHANGE || materialUuids === null || materialUuids.length !== length;
+		for ( let index = 0, length = materials.length; ( index < length ) && ! needsMaterialIndexUpdate; index ++ ) {
+
+			if ( materialUuids === null || materialUuids.length !== length ) {
```

Comment:
These conditions can be pulled out of the for loop and used in the above `needsMaterialIndexUpdate` initialization since it's not using the index value at all.